float64[] hand_init
float64[] hand_fin
float64[] obj_init
---
float64[] obj_fin
